core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/cdo/CDOTransactionBasedIdProvider.java

```diff
@@ -53,13 +53,12 @@ public CDOTransactionBasedIdProvider(final IBranchPath branchPath, final String
 	public CDOID provideCDOID(final Object idOrObject) {
 		
 		final ICDOConnection connection = ApplicationContext.getInstance().getService(ICDOConnectionManager.class).getByUuid(repositoryUuid);
-		
-		return CDOUtils.apply(new CDOTransactionFunction<CDOID>(connection, branchPath) {
-			@Override protected CDOID apply(final CDOTransaction transaction) {
-				return ((InternalCDOTransaction) transaction).provideCDOID(idOrObject);
-			}
-		});
-		
+		CDOTransaction transaction = connection.createTransaction(branchPath);
+		try {
+			return ((InternalCDOTransaction) transaction).provideCDOID(idOrObject);
+		} finally {
+			transaction.close();
```

Comment:
`LifecycleUtil.deactivate` is used somewhere else; the difference is in whether we expect `transaction` to be `null` at this point or not 😃

Reply:
Generally, I'd replace these functions with the following pattern to cover all bases:

```java
CDOView viewOrTransaction = null;
try {
  viewOrTransaction = createViewOrTransaction();
  // things are done here
} finally {
  LifecycleUtil.deactivate(viewOrTransaction);
}